src/sentry/search/events/builder/metrics.py

```diff
@@ -836,18 +889,28 @@ def get_snql_query(self) -> Request:
             )
 
             snuba_request = self.get_metrics_layer_snql_query()
-            snuba_queries, _ = SnubaQueryBuilder(
-                projects=self.params.projects,
-                metrics_query=transform_mqb_query_to_metrics_query(
-                    snuba_request.query, is_alerts_query=self.is_alerts_query
-                ),
-                use_case_id=UseCaseKey.PERFORMANCE
-                if self.is_performance
-                else UseCaseKey.RELEASE_HEALTH,
-            ).get_snuba_queries()
+
+            if self._on_demand_spec:
+                metrics_query = self._get_on_demand_metrics_query()
+
+                snuba_queries, _ = SnubaQueryBuilder(
+                    projects=self.params.projects,
+                    metrics_query=metrics_query,
+                    use_case_id=UseCaseKey.PERFORMANCE,
+                ).get_snuba_queries()
```

Comment:
_nit_: Inline this variable match structure of the other branch.

```suggestion
                snuba_queries, _ = SnubaQueryBuilder(
                    projects=self.params.projects,
                    metrics_query=self._get_on_demand_metrics_query(),
                    use_case_id=UseCaseKey.PERFORMANCE,
                ).get_snuba_queries()
```